src/contexts/PostContext.js

```diff
@@ -0,0 +1,60 @@
+import { createContext, useContext, useEffect, useState } from "react";
+
+
+const PostContext = createContext()
+
+export const usePosts = () => useContext(PostContext)
+
+const PostProvider = ({children}) => {
+
+    const [posts,setPosts] = useState([])
+
+    const fetchAndSetPosts =  async () => {
+        const response = await fetch('https://jsonplaceholder.typicode.com/posts')
+        const data = await response.json()
+        localStorage.setItem('posts',JSON.stringify(data))
+        setPosts(data)
+    }
+
+    useEffect(()=>{
+        localStorage.setItem('posts',posts)
+    },[posts])
+
+    useEffect(()=>{
+        let json = localStorage.getItem('posts')
+        if(json){
```

Comment:
if there is one statement in clock then no need of curly brackets and we can write in single line